test/one/profiler/test/TestProcess.java

```diff
@@ -0,0 +1,265 @@
+        }
+        cmd.add(test.mainClass().getName());
+        addArgs(cmd, test.args());
+        log.log(Level.FINE, "Running " + cmd);
+        ProcessBuilder pb = new ProcessBuilder(cmd).inheritIO();
+        if (test.output()) {
+            pb.redirectOutput(createTempFile("%out", null));
+        }
+        if (test.error()) {
+            pb.redirectError(createTempFile("%err", null));
+        }
+        if (!test.agentArgs().isEmpty() || test.jvmArgs().contains("-Djava.library.path=build/lib")) {
+            pb.redirectOutput(createTempFile("%pout", null));
+            pb.redirectError(createTempFile("%perr", null));
+        }
+
+        this.p = pb.start();
+        // Give the JVM some time to initialize
+        Thread.sleep(1500);
+    }
+
+    private void addArgs(List<String> cmd, String args) {
+        if (args != null && !args.isEmpty()) {
+            args = substituteFiles(args);
+            for (StringTokenizer st = new StringTokenizer(args, " "); st.hasMoreTokens(); ) {
+                cmd.add(st.nextToken());
+            }
+        }
+    }
+
+    private String substituteFiles(String s) {
+        Matcher m = filePattern.matcher(s);
+        if (!m.find()) {
+            return s;
+        }
+
+        StringBuffer sb = new StringBuffer();
+        do {
+            File f = createTempFile(m.group(1), m.group(2));
+            m.appendReplacement(sb, f.toString());
+        } while (m.find());
+
+        m.appendTail(sb);
+        return sb.toString();
+    }
+
+    private File createTempFile(String fileId, String suffix) {
+        try {
+            File f = File.createTempFile("ap-" + fileId.substring(1), suffix);
+            tmpFiles.put(fileId, f);
+            return f;
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private void clearTempFiles() {
+        for (File file : tmpFiles.values()) {
+            file.delete();
+        }
+    }
+
+    private void moveLogs() {
+        try {
+            String workingDirectory = System.getProperty("user.dir");
+            File directory = new File(workingDirectory + "/build/test/logs/" + testName);
+            directory.mkdirs();
+
+            File outDirectory = (tmpFiles.get("%pout") != null) ? tmpFiles.get("%pout") : tmpFiles.get("%out");
+            outDirectory.renameTo(new File(directory, "stdout"));   
+            
+            File errDirectory = (tmpFiles.get("%perr") != null) ? tmpFiles.get("%perr") : tmpFiles.get("%err");
+            errDirectory.renameTo(new File(directory, "stderr"));
+            
+            File profileDirectory = tmpFiles.get("%f");
+            if (profileDirectory != null) {
+                profileDirectory.renameTo(new File(directory, "profile"));
+            }
+        } catch (Throwable e) {
+            log.log(Level.WARNING, "Failed to retain logs: " + e.getMessage(), e.getStackTrace());
```

Comment:
The last argument in all `log` statements with an exception should be the exception itself (not the stack trace).